Sources/ScriptParse/ScriptParse.swift

```diff
@@ -0,0 +1,105 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ 
+ FIXME: This is a temporary alternative of the frontend implementation.
+ */
+
+import SwiftSyntax
+import TSCBasic
+import Foundation
+import ArgumentParser
+
+public struct ScriptParse: ParsableCommand {
+    @Argument var file: String
+
+    static let encoder: JSONEncoder = {
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        return encoder
+    }()
+
+    public init() {}
+}
+
+extension ScriptParse {
+    static func parse(_ path: AbsolutePath) throws -> ScriptDependencies {
+        let syntaxTree = try SyntaxParser.parse(path.asURL)
+
+        var inPackageScope = false
+        var keyStatements: [CodeBlockItemSyntax] = []
+        for statement in syntaxTree.statements {
+            if statement.statementKind == .import && inPackageScope {
+                keyStatements.append(statement)
+            } else {
+                inPackageScope = false
+            }
+            if statement.statementKind == .package
+                && statement.nextToken?.tokenKind == .importKeyword {
+                inPackageScope = true
+                keyStatements.append(statement)
+            }
+        }
+
+        var collected: [PackageDependency] = []
+
+        try keyStatements.forEach {
+            switch $0.statementKind {
+            case .package:
+                var tokens = [TokenSyntax]($0.tokens.dropFirst(2))
```

Comment:
Are these first two tokens that are dropped are known to always be `package` and `import`?

Reply:
It's `@` and `package` here, which we ensured by `if tokens.starts(with: [.atSign, .identifier("package")])`.